ansible/wait-control-host.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: Wait for Control Host to be reachable
+  hosts: ansible_control
+  gather_facts: false
+  vars_files:
+    - vars/defaults.yml
+  tasks:
+    - name: Ensure hosts are reachable
+      ansible.builtin.wait_for_connection:
+
+    # The cloud-final.service unit can run apt update, which acquires the dpkg
+    # lock and prevents other tasks from acquiring it. Wait for it to finish.
+    - name: Wait for cloud-final.service unit to complete
+      ansible.builtin.shell:
+        cmd: >-
+          systemctl is-active --quiet cloud-final.service ||
+          systemctl is-failed --quiet cloud-final.service
+      register: result
+      until: result.rc == 0
+      retries: 36
+      delay: 5
+      changed_when: false
```

Comment:
Could we not do something like this.

```
- name: Wait for cloud init to finish
  community.general.cloud_init_data_facts:
    filter: status
  register: res
  until: "res.cloud_init_data_facts.status.v1.stage is defined and not res.cloud_init_data_facts.status.v1.stage"
  retries: 50
  delay: 5
```

Taken from https://docs.ansible.com/ansible/latest/collections/community/general/cloud_init_data_facts_module.html#examples

Reply:
Possibly. I noticed that sometimes it fails, and I don't know whether that code would handle that case.

Reply:
I tested it using this module and it seems to work.